src/nock/matcher.ts

```diff
@@ -227,13 +220,34 @@ export class Matcher {
       });
     }
 
+    const contentLength = getHeaderValue(oldHeaders, newHeaders, 'content-length');
+
+    if (contentLength.old !== contentLength.new) {
```

Comment:
Are you sure we want be this strict? Won't this fail also when noise field changes? Eg id, created, etc.?

Reply:
You're right, I'll test it with demo service to be sure.

Reply:
Alright, I'll remove it, it changes also when you change the input in demo-service case.